pkg/controller/podautoscaler/replica_calculator.go

```diff
@@ -365,6 +365,12 @@ func groupPods(pods []*v1.Pod, metrics metricsclient.PodMetricsInfo, resource v1
 		if pod.DeletionTimestamp != nil || pod.Status.Phase == v1.PodFailed {
 			continue
 		}
+		// Unschedulable pods are always ignored.
+		_, scheduled := podutil.GetPodCondition(&pod.Status, v1.PodScheduled)
+		if scheduled != nil && scheduled.Status == v1.ConditionFalse && scheduled.Reason == v1.PodReasonUnschedulable {
```

Comment:
What if there is no scheduled condition? Isn't pod unscheduled in that case?
Shouldn't we also ignore pods that were scheduled but are not running yet?
Maybe we should use pod.Status.Phase instead?